linux_os/guide/system/accounts/accounts-restrictions/account_expiration/account_disable_inactivity_password_auth/bash/shared.sh

```diff
@@ -16,13 +16,17 @@ fi
 {{{ bash_ensure_pam_module_line("$PAM_FILE_PATH",
                                 'auth',
                                 'sufficient', 
-                                'pam_unix.so') }}}
+                                'pam_unix.so',
+                                '^\s*auth.*required.*pam_lastlog\.so.*') }}}
 # Ensure pam_unix.so is configured after pam_lastlog.so
 if ! grep -Pz \
     "auth\s*required\s*pam_lastlog\.so[^#]*inactive=35[\s\S]*\n\s*auth\s*sufficient\s*pam_unix\.so"\
     "$PAM_FILE_PATH"  ; then
-    PAM_LASTLOG_LINE="$(grep -oP '^\s*auth.*pam_lastlog\.so.*' $PAM_FILE_PATH)"
-    sed -i "0,/^\s*auth.*pam_unix\.so.*/i$PAM_LASTLOG_LINE" "$PAM_FILE_PATH"
+    readarray -t pam_lastlog_lines <<< $(grep -oP '^\s*auth.*pam_lastlog\.so[^#]*inactive=35.*' $PAM_FILE_PATH)
```

Comment:
The scenario is the test `wrong_order.fail.sh`,  in that test we have as initial file 
```
# Generated by authselect on Tue Jul 30 19:05:11 2024
# Do not modify this file manually.

auth        required                                     pam_env.so
auth        required                                     pam_faildelay.so delay=2000000
auth        sufficient                                   pam_unix.so nullok
auth required pam_lastlog.so inactive=35
auth        required                                     pam_deny.so

account     required                                     pam_unix.so

password    requisite                                    pam_pwquality.so
password    sufficient                                   pam_unix.so sha512 shadow nullok use_authtok
password    required                                     pam_deny.so

session     optional                                     pam_keyinit.so revoke
session     required                                     pam_limits.so
-session    optional                                     pam_systemd.so
session     [success=1 default=ignore]                   pam_succeed_if.so service in crond quiet use_uid
session     required                                     pam_unix.so
```
and after the bash remediation we have 
```
# Generated by authselect on Tue Jul 30 19:09:20 2024
# Do not modify this file manually.

auth required pam_lastlog.so inactive=35
auth        required                                     pam_env.so
auth required pam_lastlog.so inactive=35
auth        required                                     pam_faildelay.so delay=2000000
auth required pam_lastlog.so inactive=35
auth required pam_lastlog.so inactive=35
auth        sufficient                                   pam_unix.so nullok
auth required pam_lastlog.so inactive=35
auth        required                                     pam_deny.so

account     required                                     pam_unix.so

password    requisite                                    pam_pwquality.so
password    sufficient                                   pam_unix.so sha512 shadow nullok use_authtok
password    required                                     pam_deny.so
```
The automatus tool  mark that test as correct because the condition pass but not as expected.

Reply:
OK makes sense

So just please fix the shellcheck issues